lfs/proxy_test.go

```diff
@@ -0,0 +1,71 @@
+package lfs
+
+import (
+	"net/http"
+	"net/url"
+	// "os"
+	"testing"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/technoweenie/assert"
+)
+
+// --- FAIL: TestProxyFromEnv (0.00s)
+// panic: runtime error: invalid memory address or nil pointer dereference [recovered]
+// panic: runtime error: invalid memory address or nil pointer dereference
+//
+// func TestProxyFromEnv(t *testing.T) {
```

Comment:
So the relevant bit here is this:

```
github.com/github/git-lfs/lfs.TestProxyFromEnv(0xc820366000)
    /Users/jonmagic/gopath/src/github.com/github/git-lfs/lfs/proxy_test.go:29 +0x34a
```

This is telling me that the `TestProxyFromEnv()` function on `proxy_test.go:29` is failing due to a nil pointer reference. Going by the commented test line numbers, that's coming from the middle line here, because `proxyURL` is nil.

```go
proxyURL, err := proxyFromGitConfigOrEnvironment(req)
assert.Equal(t, "proxy-from-env:8080", proxyURL.Host)
assert.Equal(t, nil, err)
```

You should check `err` before the value to avoid issues like this.